### ISSUE TITLE
不動産boston

### ISSUE THREAD

**Issue by Kunihiro-ikd** [Kunihiro-ikd/data_analyze#4]
from sklearn import datasets
# ボストンの住宅価格データセットの読み込み
boston = datasets.load_boston()
# データセットの情報
print(dir(boston)) # ['DESCR', 'data', 'feature_names', 'target']

# bostonデータセットの詳細な説明
#print(boston.DESCR)

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline 
X = boston.data
y = boston.target

df_X = pd.DataFrame(X)
df_y = pd.DataFrame(y)
df_X.columns = boston.feature_names
df_y.columns = ["target"]

df_X.head()

df_y.head()

df = pd.concat([df_X, df_y], axis = 1)

df.head()

corr = df.corr()
corr

lstat = df_X.loc[:,"LSTAT"]

print("変換前のデータ型は", type(lstat))
print("変換前の配列の大きさは", lstat.shape)
lstat = lstat.values #.valuesはメソッド（関数）ではなく、属性（プロパティ）の1種で、()は不要　属性には他にも.indexや.columns
lstat = lstat.reshape(-1,1)
print("変換後のデータ型は", type(lstat))
print("変換後の配列の大きさは", lstat.shape)

plt.scatter(lstat, y)
plt.show()

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(lstat, y, test_size=0.3, random_state=0)

from sklearn.linear_model import LinearRegression
lr = LinearRegression()

lr.fit(X_train, y_train)

plt.scatter(lstat, y)                       
plt.plot(X_test, lr.predict(X_test), color='red') 
plt.title('boston_housing')
plt.xlabel('LSTAT')
plt.ylabel('target')
plt.show()        

y_train_pred = lr.predict(X_train)
y_test_pred = lr.predict(X_test)

# 性能評価
from math import sqrt
from sklearn.metrics import mean_squared_error
# 二乗平均平方根誤差(RMSE)を算出
print('RMSE Test :' + str((sqrt(mean_squared_error(y_test, y_test_pred)))))
# 学習用、検証用データに関してR^2を出力 (回帰モデルの場合score()を使うことで決定係数get)
print('R^2 Train : %.3f, Test : %.3f' % (lr.score(X_train, y_train), lr.score(X_test, y_test)))

**Comment by Kunihiro-ikd** [comment]
日本語が巨大化してる。editの時は通常サイズ。
